src/App.tsx

```diff
@@ -17,35 +21,36 @@ export const App = () => (
     >
       <div className="container">
         <div className="navbar-brand">
-          <a href="/" className="navbar-item is-active">Home</a>
-          <a href="/tabs" className="navbar-item">Tabs</a>
+          <NavLink
+            to="/"
+            className={({ isActive }) => cn('navbar-item',
+              { 'is-active': isActive })}
+          >
+            Home
+          </NavLink>
+
+          <NavLink
+            to="/tabs"
+            className={({ isActive }) => cn('navbar-item',
+              { 'is-active': isActive })}
+          >
+            Tabs
+          </NavLink>
         </div>
       </div>
     </nav>
 
     <div className="section">
       <div className="container">
-        <h1 className="title">Home page</h1>
-        <h1 className="title">Tabs page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="tabs is-boxed">
-          <ul>
-            <li data-cy="Tab" className="is-active">
-              <a href="#/">Tab 1</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 2</a>
-            </li>
-            <li data-cy="Tab">
-              <a href="#/">Tab 3</a>
-            </li>
-          </ul>
-        </div>
-
-        <div className="block" data-cy="TabContent">
-          Please select a tab
-        </div>
+        <Routes>
+          <Route path="/" element={<HomePage />} />
+          <Route path="home" element={<Navigate to="/" />} />
+          <Route path="tabs">
+            <Route index element={<TabsPage />} />
+            <Route path=":tabId" element={<TabsPage />} />
+          </Route>
+          <Route path="*" element={<NotFoundPage />} />
+        </Routes>
```

Comment:
1. Let's create a component `Layout`. 
2. Wrap all your routes to the `layout route`. 
3. This layout component should render a `Navigation` and wrap the content to the section. You can use Outlet from react-router-dom.
4. Use container div to wrap content inside your pages, to avoid using react fragments in some cases

```
  <main className="section">
    <Outlet />
  </main>
```

```suggestion
<Routes>
  <Route path="/" element={<Layout />}>
    .. homepage
    ...tabs
    .. etc
</Routes>     
```